cdap-app-fabric-tests/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -195,11 +196,13 @@ public void testLocalDatasetDeletion() throws Exception {
     Map<String, String> filesetProperties = ImmutableMap.of("anotherFoo", "anotherBar");
 
     deploy(WorkflowAppWithLocalDatasets.class, 200, Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE2);
+    ApplicationDetail applicationDetail = getAppDetails(TEST_NAMESPACE2, WorkflowAppWithLocalDatasets.NAME);
 
     File waitFile = new File(tmpFolder.newFolder() + "/wait.file");
     File doneFile = new File(tmpFolder.newFolder() + "/done.file");
 
-    ProgramId workflowId = new ProgramId(TEST_NAMESPACE2, WorkflowAppWithLocalDatasets.NAME, ProgramType.WORKFLOW,
+    ProgramId workflowId = new ProgramId(TEST_NAMESPACE2, WorkflowAppWithLocalDatasets.NAME,
+                                         applicationDetail.getAppVersion(), ProgramType.WORKFLOW,
                                          WorkflowAppWithLocalDatasets.WORKFLOW_NAME);
 
     startProgram(Id.Program.fromEntityId(workflowId), ImmutableMap.of("wait.file", waitFile.getAbsolutePath(),
```

Comment:
StartProgram calls versionless endpoint and should use reference, not id:
```
    String path = String.format("apps/%s/%s/%s/start",
                                program.getApplicationId(),
                                program.getType().getCategoryName(),
                                program.getId());
                                ```

Reply:
Also other endpoints here seems to be versionless. Why do you need a version?

Reply:
right, fixed this